src/core/net/ip6_address.cpp

```diff
@@ -145,15 +145,20 @@ bool Address::IsIidReserved(void) const
 
 void Address::SetPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength)
 {
-    uint8_t extraBits = (aPrefixLength % CHAR_BIT);
+    SetPrefix(0, aPrefix, aPrefixLength);
+}
+
+void Address::SetPrefix(uint8_t aOffset, const uint8_t *aPrefix, uint8_t aPrefixLength)
+{
+    uint8_t extraBits = aPrefixLength % CHAR_BIT;
 
-    OT_ASSERT(aPrefixLength <= sizeof(Address) * CHAR_BIT);
+    OT_ASSERT(aPrefixLength <= (sizeof(Address) - aOffset) * CHAR_BIT);
 
-    memcpy(mFields.m8, aPrefix, (aPrefixLength - extraBits) / CHAR_BIT);
+    memcpy(mFields.m8 + aOffset, aPrefix, aPrefixLength / CHAR_BIT);
 
     if (extraBits > 0)
     {
-        uint8_t index = aPrefixLength / CHAR_BIT;
+        uint8_t index = aOffset + (aPrefixLength / CHAR_BIT);
```

Comment:
Use local variable to avoid computing `aPrefixLength / CHAR_BIT` multiple times?